Adding deprecation warning for the Http package

burzum opened the pull request:
Refs #11385

burzum committed:
Adding deprecation warning for the Http package

markstory commented:
I will work on getting the tests updated over the next few days as there are quite a few uses.

markstory committed:
Fix deprecated method use in formhelper/tests.

markstory committed:
fix deprecation warning.

markstory committed:
Fix deprecated method usage in Helpers.

markstory committed:
Fix deprecated method use in View tests.

markstory committed:
Fix failing deprecated method use in ExceptionRenderer tests.

Fix a bunch of related things as they were in the call stack as well.

markstory committed:
Fix remaining tests in Error package.

burzum committed:
Fixing the tests for auth package

I had to change loginHeaders() for the Basic and Digest authenticator. I've had a short discussion with Jose about this and we agreed on doing a BC breaking change here.

burzum commented:
@markstory please take a look at this commit https://github.com/cakephp/cakephp/pull/11414/commits/3f1d4dac10358f6b87edd9038b1789da309e8554 I had a short discussion with @lorenzo about this. If you want to do it different feel free to change it.

burzum committed:
Fix ServerRequestTest cases

ADmad commented on src/Http/ServerRequest.php:
```diff
@@ -625,6 +635,11 @@ public function __call($name, $params)
     public function __get($name)
     {
+        deprecationWarning(
+            'Accessing routing parameters through __get will removed in 4.0.0. ' .
```
A message like `"Accessing '{$name}' as property has been deprecated and will be removed in 4.0.0"` might be more easily understandable to novice users.

burzum committed:
Fixing more test cases in ServerRequestTest

markstory commented:
> please take a look at this commit 3f1d4da I had a short discussion with @lorenzo about this. If you want to do it different feel free to change it.

I think what you have is reasonable for that given the constraints we're facing.

markstory committed:
Fix deprecation failures for RequestHandlerComponent.

saeideng commented on tests/TestCase/Controller/Component/RequestHandlerComponentTest.php:
```diff
@@ -542,37 +553,44 @@ public function testStartupProcessData()
      */
     public function testStartupCustomTypeProcess()
     {
-        $restore = error_reporting(E_ALL & ~E_USER_DEPRECATED);
```
please re-base 
I removed this line in other PR
`testStartupCustomTypeProcess` should not flag `@group deprecated`

markstory replied:
Thanks!

markstory committed:
Fix deprecated failures in SecurityComponent.

markstory committed:
Fix deprecation warnings in ControllerTest.

A few changes were required for Controller::redirect() to make immutable
method usage able to create the same results as the mutable methods.

markstory committed:
Fix tests for ServerRequest to no longer use deprecated methods.

markstory committed:
Add ServerRequest::withoutData()

In order to update CsrfComponent we need a new method to remove data
from the request in an immutable pattern.

markstory committed:
Update CsrfComponent to use immutable methods.

Use immutable methods in CsrfComponent. This requires changing some
protected method return values, but that is within our BC guidelines.

markstory committed:
Fix deprecated method use in CookieComponent.

markstory committed:
Fix deprecation warnings in AuthComponent tests.

markstory committed:
Fix a few tests in the Response tests.

markstory committed:
Fix deprecated methods in Http\Response

markstory committed:
Fix up tests/behavior for CorsBuilder.

Change CorsBuilder to use immutable methods. This requires a matching
change in Response to copy new headers into the current response when
using the deprecated mode of cors().

markstory committed:
Fix up tests for Response::checkNotModified.

markstory committed:
Fix remaining deprecated method use in Response test.

markstory committed:
Fix warnings in Http/Client tests.

markstory committed:
Fix deprecation warnings in Client/Auth.

markstory committed:
Fix deprecation warnings in client request/response

markstory committed:
Mark transformers as deprecated.

They will be removed in 4.0

markstory committed:
Fix remaining tests in Http package.

markstory committed:
Improve deprecation warning for Request::here()

markstory committed:
Update test controllers to use undeprecated APIs.

markstory committed:
Fix test methods with multiple requests failing to find Vendor/bin

markstory committed:
Update IntegrationTestCase tests to use httpServer.

By default always use the http/server implementation as it is not
deprecated. Add a few tests to cover the legacy dispatcher process. I've
also reverted the changes made in #10957 as they are no longer
necessary. The `getRequestTarget()` method does not return an encoded
query string so we don't need to encode it when generating security
tokens either.

saeideng commented on tests/test_app/TestApp/Controller/TestsAppsController.php:
```diff
@@ -62,9 +60,7 @@ public function redirect_to_permanent()
-        $this->response->type('json');
-
-        return $this->response;
+        return $this->response->withType('json');
```
is this true?
```php
       $this->response->type('json');

        return $this->response;
```
will change origin `response` and returns (mutable )
```php
$this->response->withType('json');
```
will no change origin `response` and returns (immutable )


I mean

```php
// value of $this->response
$responseResult=set_type()

$this->response == $responseResult //before
$this->response != $responseResult //now
```

I know this is just a testCase

markstory replied:
You're right this doesn't modify the controller's `response` property. In this situation that isn't relevant as a controller action returning a response overrides the property.

saeideng commented:
` Files changed 66 ` until now :clap:

markstory commented:
@saeideng Almost below 100 failing tests as well.

markstory committed:
Fix failing tests for cookie encryption.

markstory committed:
Deprecate the array form of Router::setRequestInfo()

This was missed in 3.5 and is a form that should not be used and we can
stop using now.

markstory committed:
Add deprecation warning for global state usage.

I missed this earlier but found it because
ServerRequest::createFromGlobals() is also deprecated.

markstory committed:
Fix deprecated method use in AssetFilter.

markstory committed:
Fix deprecated method use in RoutingFilter.

markstory committed:
Mark Dispatcher as deprecated and flag its tests.

Dispatcher is part of the deprecated Http stack. Encourage the use of
the Http\Server stack instead.

markstory committed:
DispatcherFactory is also deprecated.

Mark its tests accordingly.

markstory committed:
Fix deprecated method use.

markstory committed:
Fix tests failing due to test suite changes, and implementation changes in test classes

markstory committed:
Fix deprecated method usage in PaginatorHelper.

markstory committed:
Fix failing tests in ViewBuilder.

Can't use assertSame() as response is immutable.

ADmad commented:
:white_check_mark:  :tada:
